### ISSUE TITLE
Quarto document: "Error in dev.off() :    QuartzBitmap_Output - unable to open file",  when project located on external drive

### ISSUE THREAD

**Issue by marioem** [rstudio/rstudio#15488]
### System details

    RStudio Edition : Desktop
    RStudio Version : RStudio 2024.09.1+394 
    OS Version      : macOS 15.1.1
    R Version       : 4.4.1

    Package versions:
            tidyverse: 2.0.0
            quarto: 1.4.4
            knitr: 1.49

            quarto -v
            1.3.433

### Steps to reproduce the problem

1. Give RStudio full disk access in macOS System Settings
2. Create an RStudio project on an external drive (USB connected - Problem variant 1, or NAS - Problem variant 2)
3. Create a new quarto document, in **Visual** mode
4. Insert following code cell
````
library(tidyverse)
mtcars %>% 
  ggplot(aes(cyl, mpg)) + 
  geom_point()
````
5. Run the cell
6. Observe the result. 
Problem variant 2 (NAS drive - SMB connection) - cell code execution fails with the error:
````
Error in dev.off() : 
  QuartzBitmap_Output - unable to open file '/Volumes/Home/Dokumenty/GitHub/LSSGBHospitalWaitTimes/.Rproj.user/shared/notebooks/-HospitalWaitTimeReductionProject/1/s/cfeqomghy79s9_t/_rs_chunk_plot_001.png'
````

Problem variant 1 (USB connection) - cell code execution is successful, chart is displayed

7. In situation of Problem Variant 1 (project on USB drive) modify the code cell, after in had been run, observe now it fails
````
library(tidyverse)
mtcars %>% 
  ggplot(aes(cyl, mpg)) + 
  geom_point() +
  labs(title = "Test")

Warning in gzfile(file, "wb") :
  cannot open compressed file '/Volumes/podBase/RStudio/LSSGBHospitalWaitTimes/.Rproj.user/shared/notebooks/261D7241-HospitalWaitTimeReductionProject/1/49E84C33b65ed375/ch9uvedo3whvn_t/eaba06e3a56a46f18eb334f4984fa667.snapshot', probable reason 'No such file or directory'
Error in gzfile(file, "wb") : cannot open the connection
Error in dev.off() : 
  QuartzBitmap_Output - unable to open file '/Volumes/podBase/RStudio/LSSGBHospitalWaitTimes/.Rproj.user/shared/notebooks/261D7241-HospitalWaitTimeReductionProject/1/49E84C33b65ed375/ch9uvedo3whvn_t/_rs_chunk_plot_001.png'
````
8. Switch to **Source** view and re-run the code cell, modify the code re-run again, now it works every time, when project is located on USB drive, but still fails every time when project is located on a NAS drive.

### Describe the problem in detail

See above

### Describe the behavior you expected

Rendering of in-line charts in qmd documents works independently which type of drive the project is located on

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.posit.co/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kevinushey** [comment]
Thanks for the bug report -- you might try customizing the `.Rproj.user` folder location.

<img width="672" alt="Image" src="https://github.com/user-attachments/assets/b864a3c0-66d0-4bc6-8305-05f6313acd4d" />

In particular, you might have better luck using a local filesystem in this scenario.

**Comment by marioem** [comment]
Hi,

thanks for workaround.

The reason I'm using network drive and not local filesystem is to have access to my projects from whatever computer I happen to be working at, apart from having no immediate storage space limitations.

I changed .Rproj.user folder location to my iCloud drive, so I kind of keep my cross-machine project portability, but this is is not as elegant as not having to maintain a separate foder structure for .Rproj.user on iCloud drive.

BRs,

Mariusz

**Comment by marioem** [comment]
No, .Rproj.user on iCloude drive is not a reliable solution, the same error appears usually upon second execution of a given code cell, goes away for a while after restarting R session, and then the phenomenon repeats.
